Source/Core/Common/MemArena.cpp

```diff
@@ -67,12 +80,47 @@ static int AshmemCreateFileMapping(const char* name, size_t size)
 }
 #endif
 
+MemArena::MemArena()
+{
+#ifdef _WIN32
+  // Check if VirtualAlloc2 and MapViewOfFile3 are available, which provide functionality to reserve
+  // a memory region no other allocation may occupy while still allowing us to allocate and map
+  // stuff within it. If they're not available we'll instead fall back to the 'legacy' logic and
+  // just hope that nothing allocates in our address range.
+  if (!IsApiSetImplemented("api-ms-win-core-memory-l1-1-6"))
+    return;
+
+  HMODULE handle = LoadLibraryW(L"api-ms-win-core-memory-l1-1-6.dll");
+  if (!handle)
+    return;
+
+  m_address_VirtualAlloc2 = GetProcAddress(handle, "VirtualAlloc2FromApp");
+  m_address_MapViewOfFile3 = GetProcAddress(handle, "MapViewOfFile3FromApp");
+  if (!m_address_VirtualAlloc2 || !m_address_MapViewOfFile3)
```

Comment:
Will there be a problem if only one of them is null? Or is it even possible for one of them to be null?

Reply:
It seems unlikely for that to ever be the case, but either way, having one without the other isn't useful, so I just fall back to the old code if either is not here.

Reply:
I got confused by your if-guards, perhaps it's worth adding a note or a bool based on the handle to clarify how it is checked.

Reply:
Now it's very clear.